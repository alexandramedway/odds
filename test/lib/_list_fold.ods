do nums = [1, 2, 3, 4, 5]
do xor = (a, b) -> return (a || b) && !(a && b)

do print(list_fold((a, x) -> return a + x, 0, []))
do sum = list_fold((a, x) -> return a + x, 0, nums)
do print(sum)

do print(list_fold(xor, true, [false, true, false, true]))
do print(list_fold(xor, true, [false, true, true, true]))
