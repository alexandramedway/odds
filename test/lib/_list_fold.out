0
15
True
False
